plugins/destination/postgresql/client/migrate.go

```diff
@@ -48,165 +167,193 @@ func (c *Client) Migrate(ctx context.Context, tables schema.Tables) error {
-				stalePks = append(stalePks, pk)
-			}
+func (c *Client) alterColumnCockroachDB(ctx context.Context, tableName string, column schema.Column) error {
+	columnName := pgx.Identifier{column.Name}.Sanitize()
+	columnType := c.SchemaTypeToPg(column.Type)
+	// try alter column in place first
+	sql := "alter table " + tableName + " alter column " + columnName + " type " + columnType
+	if column.CreationOptions.NotNull {
+		sql += " set not null"
+	}
+	if _, err := c.conn.Exec(ctx, sql); err != nil {
+		c.logger.Warn().Err(err).Str("table", tableName).Str("column", column.Name).Msg("Column type changed in place failed.")
+	}
+
+	sql = "alter table " + tableName + " drop column " + columnName
+	// right now we will drop the column and re-create. in the future we will have an option to automigrate
+	if _, err := c.conn.Exec(ctx, sql); err != nil {
+		return fmt.Errorf("failed to drop column %s.%s: %w", column.Name, tableName, err)
+	}
+	sql = "alter table " + tableName + " add column " + columnName + " " + columnType
+	if column.CreationOptions.NotNull {
+		sql += " not null"
+	}
+	if _, err := c.conn.Exec(ctx, sql); err != nil {
+		var pgErr *pgconn.PgError
+		if !errors.As(err, &pgErr) {
+			return fmt.Errorf("failed to add column %s.%s: %w", tableName, column.Name, err)
+		}
+		// this means the column contains null value and we cannot recreate it without knowing what is the default
+		// This is usually the case when the column is a primary key
+		// the user will have to drop and recreate the table
+		if pgErr.Code == "23502" {
+			return fmt.Errorf("column %s.%s contains null values, please drop the table manually via 'drop table %s'", tableName, column.Name, tableName)
+		}
+		// this is a weird CockroachDB error code that means we need to retry.
+		if pgErr.Code != "55000" {
+			return fmt.Errorf("failed to add column %s.%s with pgerror %s: %w", tableName, column.Name, pgErrToStr(pgErr), err)
+		}
+		if _, err := c.conn.Exec(ctx, sql); err != nil {
+			return fmt.Errorf("failed to retry adding column %s.%s with pgerror %s: %w", tableName, column.Name, pgErrToStr(pgErr), err)
 		}
 	}
-	return stalePks
+	return nil
 }
 
-func (c *Client) autoMigrateTable(ctx context.Context, table *schema.Table) error {
-	var err error
-	var pgColumns *pgTableColumns
-	var pgPKs map[string]bool
-
-	// create the new column as it doesn't exist
-	tableName := pgx.Identifier{table.Name}.Sanitize()
-	if pgColumns, err = c.getPgTableColumns(ctx, table.Name); err != nil {
-		return fmt.Errorf("failed to get table %s columns types: %w", table.Name, err)
+func (c *Client) alterColumnPg(ctx context.Context, tableName string, column schema.Column) error {
+	columnName := pgx.Identifier{column.Name}.Sanitize()
+	columnType := c.SchemaTypeToPg(column.Type)
+	// try alter column in place first
+	sql := "alter table " + tableName + " alter column " + columnName + " type " + columnType
+	if column.CreationOptions.NotNull {
+		sql += " set not null"
```

Comment:
```suggestion
		sql += ", ALTER " + columnName + " set not null"
```

Otherwise this always fail. Also Should we return early when it's successful so we don't need to drop/add it later one?